odoo/tools/split_table.py

```diff
@@ -0,0 +1,35 @@
+from lxml.etree import Element
+
+def copy_attributes(src, dst, exclude=()):
+    """
+    Copies all attributes from src to dst. If exclude is not None, the
+    attributes in exclude are not copied.
+
+    :param src: The source element
+    :param dst: The destination element
+    :param exclude: A list of attributes that should not be copied
+    """
+    for key, value in src.attrib.items():
+        if key not in exclude:
+            dst.set(key, value)
+
+def split_table(etree, max_rows):
+    """
+    Walks through the etree and splits tables with more than max_rows rows into
+    multiple tables with max_rows rows.
+
+    This function is needed because wkhtmltopdf has a exponential memory growth
+    when processing tables with many rows. This function is a workaround for
+    this problem.
+
+    :param etree: The etree to process
+    :param max_rows: The maximum number of rows per table
+    """
+    tables_to_split = [t for t in etree.iter('table') if len(t) > max_rows]
+    for table in tables_to_split:
+        sibling = Element('table')
+        sibling.extend(table[max_rows:])
+        copy_attributes(table, sibling, exclude=('id'))
```

Comment:
```suggestion
        copy_attributes(table, sibling, exclude=('id',))
```
`('id')` is a string, you need tuple here. `exclude=('id')` works by accident 😄